### PR TITLE
Cron support for 'L' and 'LW' in expression daysOfMonth

### PR DESCRIPTION
**Pull request by jafin**
- Add cron parser support for `'L'` and `'LW'` in expression combinations daysOfMonth (#1939) (#1288)
- There is a limit of only one 'L' in the expression.  ie 
`"0 15 10 7,15,L,L-2 * ?"` will fail  but 
`"0 15 10 7,15,L-2 * ? "` will pass.

### CODE REVIEW

File: src/Quartz.Tests.Unit/CronExpressionTest.cs [hunk 1]
```diff
@@ -487,7 +509,7 @@ public void TestGetTimeAfter_QRTZNET149()
         }
 
         [Test]
-        public void TestQRTZNET152()
+        public void TestQRTZNET152_Nearest_Weekday_Expression_W_Does_Not_Work_In_CronTrigger()
```
Comment (jafin) [inline src/Quartz.Tests.Unit/CronExpressionTest.cs:512]: Renamed to better understand intent of test

**Comment by lahma** [comment]
Can you also update change log so that it's easier to keep track of the changes?

**Review by lahma — APPROVED** [review]
This required surprisingly lot of changes, but I'm taking somewhat a leap of faith as review is quite difficult for this changeset. Thanks for implementing this!

**Comment by jafin** [comment]
@lahma a lot of the change was refactoring, the cyclomatic complexity of a few of the methods very high, I found it quite challenging to work through, noting that it's been that way since the java port. 

All tests are passing, plus more were added, but I'll see if I can get some more coverage in there for confidence.

In future would it be easier to review, if these types of changes were presented as smaller commits rather than squashed for review?

**Pull request merged**